internal/services/providers/gke/gke.go

```diff
@@ -48,6 +58,49 @@ func (p *Provider) RegisterCluster(ctx context.Context, client castai.Client) (*
 	}, nil
 }
 
+func (p *Provider) clusterAutodiscovery() (*config.GKE, error) {
+	var err error
+	cfg := &config.GKE{}
+	if envCfg := config.Get().GKE; envCfg != nil {
+		cfg = envCfg
+	}
+
+	if cfg.ProjectID == "" {
+		cfg.ProjectID, err = p.metadata.GetProjectID()
+		if err != nil {
+			return nil, failedAutodiscovery(err, "GKE_PROJECT_ID")
+		}
+	}
+
+	if cfg.ClusterName == "" {
+		cfg.ClusterName, err = p.metadata.GetClusterName()
+		if err != nil {
+			return nil, failedAutodiscovery(err, "GKE_CLUSTER_NAME")
+		}
+	}
+
+	if cfg.Region == "" {
+		cfg.Region, err = p.metadata.GetRegion()
+		if err != nil {
+			return nil, failedAutodiscovery(err, "GKE_REGION")
+		}
+	}
+
+	if cfg.Location == "" {
+		cfg.Location, err = p.metadata.GetLocation()
+		if err != nil {
+			// Not required during phase 1 registration. Will fail if not able to get during Phase 2 registration.
+			p.log.Warningf("getting cluster location during autodiscovery: %w", err)
```

Comment:
Location is required for phase 1 as well, I would treat it same as region - return err if failed